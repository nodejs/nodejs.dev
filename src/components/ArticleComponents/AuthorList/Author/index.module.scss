.link {
  &:hover span {
    transform: scale(1.1);
  }

  span {
    background-size: contain;
    border: 2px solid var(--brand-light);
    border-radius: 100%;
    display: block;
    height: 30px;
    margin-top: 5px;
    transition: all 0.2s ease-in-out;
    width: 30px;
  }
}
