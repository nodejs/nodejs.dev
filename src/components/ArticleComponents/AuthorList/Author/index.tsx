import React from 'react';
import { injectIntl, WrappedComponentProps } from 'react-intl';
import styles from './index.module.scss';
import authorPlaceholderImg from '../../../../images/placeholder-img.png';

interface Props {
  index: number;
  username: string;
  size: string;
}

const Author = ({
  index,
  username,
  size = '64',
  intl,
}: Props & WrappedComponentProps) => {
  // Clean up username and build links.
  const githubUserName = username.trim();
  const githubLink = `https://github.com/${githubUserName}`;
  const githubImgLink = `https://github.com/${githubUserName}.png?size=${size}`;
  
  const authorImg = {
    backgroundImage: `url(${githubImgLink}), url(${authorPlaceholderImg})`,
  };

  // If it's the first author then no margin left.
  const mleft = index === 0 ? { marginLeft: 0 } : {};

  const translation = intl.formatMessage(
    { id: 'components.author.githubLinkLabel' },
    { username }
  );

  return (
    <li>
      <a
        className={styles.link}
        href={githubLink}
        aria-label={translation}
        key={username}
        target="_blank"
        rel="noopener noreferrer"
        style={mleft}
      >
        <span
          style={authorImg}
        />
      </a>
    </li>
  );
};

export default injectIntl(Author);
