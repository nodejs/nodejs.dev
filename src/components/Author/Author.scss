.list {
  list-style: none;
}

.link {
  margin-left: 0.5rem;
  &:hover img {
    transform: scale(1.1);
  }
}

.img {
  height: 30px;
  width: 30px;
  margin-top: 5px;
  border-radius: 100%;
  border: 1px solid var(--brand-light);
  transition: all 0.2s ease-in-out;
}
