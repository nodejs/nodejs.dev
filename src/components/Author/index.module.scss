.link {
  &:hover img {
    transform: scale(1.1);
  }

  img {
    border: 2px solid var(--brand-light);
    border-radius: 100%;
    height: 30px;
    margin-top: 5px;
    transition: all 0.2s ease-in-out;
    width: 30px;
  }
}
