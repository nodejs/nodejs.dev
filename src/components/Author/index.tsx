import React from 'react';
import './Author.scss';

interface Props {
  index: number;
  username: string;
  size: string;
}

const Author = ({
  index,
  username,
  size = '64',
}: Props): null | JSX.Element => {
  if (!username) {
    return null;
  }

  // Clean up username and build links.
  const githubUserName = username.trim();
  const githubLink = `https://github.com/${githubUserName}`;
  const githubImgLink = `https://github.com/${githubUserName}.png?size=${size}`;

  // If it's the first author then no margin left.
  const mleft = index === 0 ? { marginLeft: 0 } : {};

  return (
    <li>
      <a
        className="link"
        href={githubLink}
        title={username}
        key={username}
        target="_blank"
        rel="noopener noreferrer"
        style={mleft}
      >
        <img className="img" src={githubImgLink} alt={username} />
      </a>
    </li>
  );
};

export default Author;
