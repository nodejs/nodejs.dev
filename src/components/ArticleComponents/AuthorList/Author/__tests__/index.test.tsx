import React from 'react';
import { render } from '@testing-library/react';
import Author from '..';

describe('Author component', () => {
  it('renders correctly', () => {
    const username = 'test-author';
    const { container } = render(
      <Author index={1} username={username} size="60" />
    );
    expect(container).toMatchSnapshot();
  });

  it('does not render without a username', () => {
    const { container } = render(<Author index={0} username="" size="" />);
    expect(container).toMatchSnapshot();
  });
});
