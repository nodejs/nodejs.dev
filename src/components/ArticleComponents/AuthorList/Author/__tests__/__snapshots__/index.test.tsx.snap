// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author component does not render without a username 1`] = `
<div>
  <li>
    <a
      aria-label=" Github - opens in new tab"
      class="link"
      href="https://github.com/"
      rel="noopener noreferrer"
      style="margin-left: 0px;"
      target="_blank"
    >
      <img
        alt=""
        src=""
      />
    </a>
  </li>
</div>
`;

exports[`Author component renders correctly 1`] = `
<div>
  <li>
    <a
      aria-label="test-author Github - opens in new tab"
      class="link"
      href="https://github.com/test-author"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt=""
        src=""
      />
    </a>
  </li>
</div>
`;
